src/serac/physics/materials/parameterized_thermal_functional_material.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2019-2022, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file thermal_functional_material.hpp
+ *
+ * @brief The material and load types for the thermal functional physics module
+ */
+
+#pragma once
+
+#include "serac/physics/materials/thermal_functional_material.hpp"
+
+/// ThermalConductionFunctional helper structs
+namespace serac::Thermal {
+
+/// Linear isotropic conductor with a parameterized conductivity
+class ParameterizedLinearIsotropicConductor {
+public:
+  /**
+   * @brief Construct a new Parameterized Linear Isotropic Conductor object
+   *
+   * @param density Density of the material (mass/volume)
+   * @param specific_heat_capacity Specific heat capacity of the material (energy / (mass * temp))
+   * @param conductivity_offset Thermal conductivity offset of the material (power / (length * temp)). This is
+   * added to the parameter value to get the total conductivity.
+   */
+  ParameterizedLinearIsotropicConductor(double density = 1.0, double specific_heat_capacity = 1.0,
+                                        double conductivity_offset = 0.0)
+      : density_(density), specific_heat_capacity_(specific_heat_capacity), conductivity_offset_(conductivity_offset)
+  {
+    SLIC_ERROR_ROOT_IF(conductivity_offset_ < 0.0,
+                       "Conductivity must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(specific_heat_capacity_ < 0.0,
+                       "Specific heat capacity must be positive in the linear isotropic conductor material model.");
+  }
+
+  /**
+   * @brief Thermal material response operator
+   *
+   * @tparam T1 Spatial position type
+   * @tparam T2 Temperature type
+   * @tparam T3 Temperature gradient type
+   * @tparam T4 Parameter type
+   * @param temperature_gradient The spatial gradient of the temperature (d temperature dx)
+   * @param parameter The user-defined parameter used to compute the conductivity (total conductivity = conductivity
+   * offset + parameter)
+   * @return The density, specific heat capacity, and heat flux of the material.
+   */
+  template <typename T1, typename T2, typename T3, typename T4>
+  SERAC_HOST_DEVICE auto operator()(const T1& /* x */, const T2& /* temperature */, const T3& temperature_gradient,
+                                    const T4& parameter) const
+  {
+    using FluxType = decltype((conductivity_offset_ + parameter) * temperature_gradient);
+
+    return MaterialResponse<double, double, FluxType>{
```

Comment:
Did this explicit type specification end up being required? I remember we tried this when debugging that earlier problem